.github/workflows/ci.yml

```
@@ -23,8 +86,33 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Determine if build is allowed
+        id: should_build
+        shell: bash
+        run: |
+          is_fork="${{ needs.context.outputs.is_fork == 'true' }}"
+          is_default_branch="${{ needs.context.outputs.is_default_branch == 'true' }}"
+          is_dependabot="${{ needs.context.outputs.is_dependabot == 'true' }}"
+
+          # Default behaviour is to build images for any CI.yml run
+          should_build="true"
+
+          # Never run the build on a fork. Forks lack sufficient permissions
+          # to access secrets or push artifacts
+          if [ $is_fork == 'true' ]; then
+            should_build="false"
+          fi
+
+          # Dependabot PRs are treated as if they are from forks (see above)
+          if [ $is_dependabot = 'true' && $is_default_branch = 'false']; then
```

As above but also double equal sign for consistency 

```suggestion
          if [ "$is_dependabot" == 'true' && "$is_default_branch" == 'false']; then
```

?